remote.go

```diff
@@ -454,6 +454,25 @@ func (o *Remote) PushUrl() string {
 	return C.GoString(C.git_remote_pushurl(o.ptr))
 }
 
+func (c *RemoteCollection) Rename(remote, newname string) error {
+	cproblems := C.git_strarray{}
+
+	cnewname := C.CString(newname)
+	defer C.free(unsafe.Pointer(cnewname))
+	cremote := C.CString(remote)
+	defer C.free(unsafe.Pointer(cremote))
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ret := C.git_remote_rename(&cproblems, c.repo.ptr, cremote, cnewname)
+	if ret < 0 {
+		return MakeGitError(ret)
+	}
+	C.git_strarray_free(&cproblems)
```

Comment:
We're just throwing away the issues we found here, which makes it impossible for a user of the bindings to tell the user about unexpected references they might find. We have to return this as well.